src/seedu/addressbook/data/person/Name.java

```diff
@@ -55,6 +56,37 @@ public boolean equals(Object other) {
                 || (other instanceof Name // instanceof handles nulls
                 && this.fullName.equals(((Name) other).fullName)); // state check
     }
+    
+    public boolean isSimilar(Name other){
+    	String myName = this.fullName.toLowerCase();
+    	String theirName = other.fullName.toLowerCase();
+    	
+    	if (myName == null || theirName == null)
+    		return false;
+    	
+    	List<String> myWords = Arrays.asList(myName.split("\\s+"));
+    	List<String> theirWords = Arrays.asList(theirName.split("\\s+"));
+
+    	System.out.println(myWords);
+    	
+    	if (checkIfSameWordOccurs(theirWords, myWords))
+    		return true;
+    	
```

Comment:
Indentation problem here? Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.